### PR TITLE
feat: Add authentication option for snowflake connector

### PR DESCRIPTION
**Pull request by lsaiken**
Signed-off-by: Lisa Aiken <laiken@manifold.ai>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by felixwang9817** [comment]
hey @lsaiken thanks for the PR! everything lgtm except there's a lint error that you'll need to fix

you can find more about how we handle linting [here](https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#code-style--linting)

**Comment by sfc-gh-madkins** [comment]
@lsaiken did you run into any issues of having to authenticate an excessive amount?

See https://docs.snowflake.com/en/user-guide/admin-security-fed-auth-use.html#using-connection-caching-to-minimize-the-number-of-prompts-for-authentication-optional

File: sdk/python/feast/infra/offline_stores/snowflake.py [hunk 1]
```diff
@@ -91,6 +91,9 @@ class SnowflakeOfflineStoreConfig(FeastConfigBaseModel):
     schema_: Optional[str] = Field(None, alias="schema")
     """ Snowflake schema name """
 
+    authenticator_: Optional[str] = Field("", alias="authenticator")
```
Comment (sfc-gh-madkins) [inline sdk/python/feast/infra/offline_stores/snowflake.py:94]: Was "authenticator" a reserved word?

Comment (lsaiken) [reply sdk/python/feast/infra/offline_stores/snowflake.py:94]: No, it's not a reserved keyword. I'll make the adjustment.

File: sdk/python/feast/infra/offline_stores/snowflake.py [hunk 2]
```diff
@@ -91,6 +91,9 @@ class SnowflakeOfflineStoreConfig(FeastConfigBaseModel):
     schema_: Optional[str] = Field(None, alias="schema")
     """ Snowflake schema name """
 
+    authenticator: Optional[str] = ""
```
Comment (sfc-gh-madkins) [inline sdk/python/feast/infra/offline_stores/snowflake.py:94]: Got it on excessive authentication.

See following error in integration test:

"E       snowflake.connector.errors.DatabaseError: 390139 (08001): None: Failed to connect to DB: ***.snowflakecomputing.com:443, The specified authenticator is not accepted by your Snowflake account configuration.  Please contact your local system administrator to get the correct URL to use."

My guess is your code is always passing the authenticator kwarg, when it should only be passed if specified

authenticator: Optional[str] = ""
vs
authenticator: Optional[str] = None

would be my first guess

**Comment by felixwang9817** [comment]
thanks @sfc-gh-madkins for the help here!

**Comment by lsaiken** [comment]
thanks @sfc-gh-madkins and @felixwang9817!

**Pull request merged**